docs/messaging/messaging-spans.md

```diff
@@ -237,22 +237,26 @@ into a message either from a "Create" span or as a custom creation context.
 
 #### Consumer spans
 
-"Deliver" spans SHOULD be created for operations of passing messages to the
-application when those operations are not initiated by the application code
-(push-based scenarios). A "Deliver" span covers the duration of such an
-operation, which is usually a callback or handler.
-
 "Receive" spans SHOULD be created for operations of passing messages to the
 application when those operations are initiated by the application code
 (pull-based scenarios).
 
-"Deliver" or "Receive" spans MUST NOT be created for messages that are
+"Process" spans SHOULD be created for operations of passing messages to the
+application when those operations are not initiated by the application code
+(push-based scenarios). Such "Process" span covers the duration of such an
+operation, which is usually a callback or handler.
+
+"Process" spans MAY be created in addition to "Receive" spans for pull-based
+scenarios for operations of processing messages. Such spans could be created by
+application code, or by abstraction layers built on top of messaging SDKs.
+
+"Receive" or "Process" spans MUST NOT be created for messages that are
 pre-fetched or cached by messaging libraries or SDKs until they are forwarded
 to the caller.
```

Comment:
So I agree with this approach, that the "process" span should only encapsulate the actual processing of the message by the callback in push based scenarios. As mentioned in the meeting, Google Pub/Sub client libraries do additional processing / transformations on the message prior to calling the user callback that we're interested in tracing as internal spans. In this case, the pre-processing spans would link to the "create" span, rather than the "process" span. Our process span would actually be a child of the pre-process span. I'm wondering if that conflicts with the diagram below (where the process span is linked to the create span).

Here's an example of what we have so far. The screenshot shows the trace of a single message produced and consumed, by a topic named `otel-topic` and consumer named `otel-sub`.
![image](https://github.com/open-telemetry/semantic-conventions/assets/9397363/be8ee1fa-c169-47e6-ae03-0ccc75edd42e)

Reply:
That looks good, you have the required "Create" and "Process" spans present.

The "Process" span should definitely link to the "Create" span. It can still be child of the "subscribe" span.

For naming the "Process" span, we recommend `<destination name> <operation name>`, which in your case should be `otel-topic process` I think?

Reply:
Yeah adding additional links sounds fine.